### ISSUE TITLE
Ukrainian filter

### ISSUE THREAD

**Issue by etircopyh** [uBlockOrigin/uBlock-issues#2692]
### Prerequisites

- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uAssets/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] I am running the latest version of uBlock Origin (uBO).

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

So, to be short, now we have Ukrainian specific filter, NOT a bloatpile that is ru adlist. The article: https://adguard.com/en/blog/ukrainian-filter.html
Raw link for uBlock filter: https://filters.adtidy.org/extension/ublock/filters/23.txt

Also there's Ukrainian Malicious URL Blocklist: https://github.com/braveinnovators/url-blocklist which could certainly be useful for a lot of people at these trying times, perhaps it could be added under a dropdown like polish filters.

### URL(s) where the issue occurs.

-

### Configuration

-

**Comment by gorhill** [comment]
When I import _AdGuard Ukrainian_ and search for `.ru`, I get over 2,000 filters, even more than the count for `.ua`, so it doesn't look like the list targets only `.ua` sites.

A long time ago it was decided by AdGuard and RuAd List that uBO should stick to only one RU list because mixing both was causing issues. I am assuming this is still the case, so having potentially both _AdGuard Ukraine_ used along _RuAd List_ would cause issues as per past decision.

Maybe @dimisa-RUAdList can provide more insights.

I need more input before making a decision.

**Comment by dportvine** [comment]
@etircopyh 
> Perhaps, we'll see a proper Ukrainian filter some day only when it will be done by Ukrainians for Ukrainians...

https://github.com/ukrainianfilters/lists

You can join in and make your contribution. Maybe in a few years something will come of it. The main problem is feedback. In RU AdList, I can report through Adblock Plus.

**Comment by koorool** [comment]
> The goal of making the Russian and Ukrainian filters independent of each other seems to me a non-trivial task. There will be a large number of intersections and repeating rules.
> 
> For several centuries Russia and Ukraine were one country. During this time, a huge number of cultural, family and business ties have been formed. At the moment, only a little less than half of my relatives are residents and citizens of Ukraine, while the other part and I myself are residents and citizens of Russia. And my case is by no means unique. To make the current interpenetration insignificant, the efforts of more than one generation will be required. And I hope that never happens.
> 
> As I can see from Adblock Plus reports statistics, Ukrainians visit Russian sites, and Russians visit Ukrainian sites. Ukrainian resources continue to cooperate with Russian advertising networks, despite the current conflict.
> 
> I may be wrong, but I believe the guys from AdGuard created a project of a separate Ukrainian filter in order to get the possibility of more flexible manipulation in the conditions of MV3 restrictions.
> 
> In any case, we do not plan to debug for possible conflicts, and do not even discuss it.

This is clearly a political and pro-russian, imperial position. Not a technical issue that could be resolved.

Our freedom and independence is very important to us.

> A long time ago it was decided by AdGuard and RuAd List that uBO should stick to only one RU list because mixing both was causing issues. I am assuming this is still the case, so having potentially both AdGuard Ukraine used along RuAd List would cause issues as per past decision.

1. As a user from Ukraine, I will not be enabling RuAd list together with the dedicated Ukrainian list. But I want an option to have my own dedicated Ukrainian list with only rules that apply to me. RuAd list contains more than 13k .ru websites and just a few Ukrainian rules. If they need them, they could keep them.
2. Many .ru resources are blocked in Ukraine after 2016 annexation of Crimea and Eastern regions and 2022 full-scale invasion of Ukraine. Many .ua resources are blocked in Russia.
3. All discussions on RuAd forums happen in Russian. Not all, and now fewer and fewer, Ukrainians speak Russian. Even less want to speak Russian.
4. You mentioned problems of AdGuard list. Please consider https://github.com/ukrainianfilters/lists that is of much better quality. Many approved regional filter lists are much simpler and more outdated than this list so it should be OK by your metrics.

**Comment by gorhill** [comment]
Do **not** bring politics or geopolitics here, that is not how decision are made here. More of this and I will lock the discussion.

> I want an option to have my own dedicated Ukrainian list

You have this option: use `Import...` in Filter lists pane.

> Please consider https://github.com/ukrainianfilters/lists that is of much better quality

It was created only ~4 months ago, we need a list to have been maintained for a much longer time to qualify as a candidate for stock lists.

The bottom line is that volunteers here are the ones dealing with filter issues, and some AdGuard lists have interfered negatively in the past when enabled along RUAdList. If I were to add AdGuard Ukraine as stock list, it would be enabled by default for `uk` language along RUAdList and this could add a burden on volunteers -- this was the reason for declining back then.

Whether we should re-consider adding AdGuard Ukraine to stock lists, I will leave the decision to filter list maintainers, I am not involved enough with filter lists issues to decide this on my own.

**Comment by dportvine** [comment]
@ameshkov

Hello, can you help resolve this issue? I still see a lot of `.ru` rules. The issue of a separate regional list for Ukraine is more relevant than ever.

<details>
<summary>Screen</summary>

![2024-12-10_195341](https://github.com/user-attachments/assets/ac33bd78-537e-4e45-94ac-83655502bda9)
</details>

@serhiyguryev

>Please consider https://github.com/ukrainianfilters/lists that is of much better quality

>It was created only ~4 months ago, we need a list to have been maintained for a much longer time to qualify as a candidate for stock lists.

**Comment by serhiyguryev** [comment]
The Ukrainian Filters collection has been created and maintained since [2023](https://github.com/braveinnovators/ukrainian-security-filter). The Ukrainian Security Filter was the first to be created, and work on the rest of the filters began in [February 2024](https://github.com/serhiyguryev/ukrainian-filters). So I don't quite understand where the information that “It was created only ~4 months ago” came from.

@gorhill, what prevents from adding our filters, which are created by a team that lives in Ukraine and understands what the ukrnet is, unlike the developers of the above filters?

**Comment by gorhill** [comment]
> I don't quite understand where the information that “It was created only ~4 months ago” came from.

https://github.com/ukrainianfilters/lists/graphs/commit-activity

![image](https://github.com/user-attachments/assets/9a2d1ce1-98b8-4a0d-a9d4-325cdfbabe3e)

**Comment by freezer2022** [comment]
There is also the old repository name https://github.com/serhiyguryev/ukrainian-filters/graphs/commit-activity with 6 months of developement:

![scr](https://github.com/user-attachments/assets/7808dd84-5dd2-4ae2-962c-8bfc57c493e1)

which in total gives 10 months from the both repositories.

**Comment by serhiyguryev** [comment]
> https://github.com/ukrainianfilters/lists/graphs/commit-activity

@gorhill, I have provided web links that prove that the filter collection started in 2023. These links are also available on the [project page](https://github.com/ukrainianfilters/lists/), so it's easy to check. What you are showing is the project being moved to a separate repository.

**Comment by gorhill** [comment]
> What you are showing is the project being moved to a separate repository.

Sure, you clarified that _after_ I [posted my findings](https://github.com/uBlockOrigin/uBlock-issues/issues/2692#issuecomment-2531885165).

[You said](https://github.com/uBlockOrigin/uBlock-issues/issues/2692#issuecomment-2532621165):

> I don't quite understand where the information that “It was created only ~4 months ago” came from.

The screenshot I posted was to clarify where the information came from when I wrote [my comment](https://github.com/uBlockOrigin/uBlock-issues/issues/2692#issuecomment-2531885165).


Now regarding <https://raw.githubusercontent.com/ukrainianfilters/lists/main/ads/ads.txt>:

- I see a lot of `...:remove()` in your list. what is the rationale for these?
- `||*content-loader.com/*^$script` etc.: why the wildcard?
- `://*.*/user.php?mod=do&act=login$document,xmlhttprequest` etc.: why the wildcards?
- In your list I count 107 instances of `.ua`, while I count 1,721 in RUAdList: doesn't this suggest that at the moment RUAdList is more likely to fulfill filtering expectations of users?

If I decide to include the Ukraine list as a stock list, I will remove `uk` from RUAdList so that users do not end up with two regional lists which might interfere. Because of this I need to know that the Ukraine list fulfill the filtering expectations of users vs. when RUAdList is used.

**Comment by freezer2022** [comment]
> gorhill: * In your list I count 107 instances of `.ua`, while I count 1,721 in 

I don't know where does 107 come from, as it shows 429: [screenshoot](https://github.com/user-attachments/assets/9290ef5b-525a-4dbd-9943-edd98f24616b).
But yes, even despite it, RU AdList still has much more `.ua` filters/domains.

> gorhill: instances of `.ua`

By the way, instances counted in _uBO - Asset viewer_ are lines/filters, and not occurences (domains), and since a line/filter can contain unknown/random number of `ua.` occurences (domains), it's better to count `.ua` occurences (domains), than instances (lines/filters), to get more realistic/stable results, it gives the RU ADList even more advantage:

| Filter list | .ua Instances<br> (lines/filters) | .ua Occurences<br> (domains) |
|--------|--------|--------|
| Ukrainian Ad Filter | 429 | 511 |
| RU AdList | 1721 | 2827 |
| How much more: | x4 | x5.5 | 

By the way, another factor is how many domains are alive and how many are dead/parked in both filter lists. Someone would have to check it out. By using this tool for example: https://github.com/funilrys/PyFunceble (but it can only find dead domains, and not parked domains (this feature is still not implemented)).

**Comment by freezer2022** [comment]
> serhiyguryev : 1,721 domains?

Lines/filters, not domains. Domains count is 2827, like already mentioned in the table.
(Just a small technical note, not that it changes the context of your statement)

> serhiyguryev : I also don't understand why users from Poland have a choice between two filters, while users from Ukraine, with only two filters made specifically for Ukraine, don't have the opportunity to choose the one that suits them.

These are two mutually exclusive POL lists. One is for ads, the another for anti-adblocks. In the past, both of them were in the same file, as a single POL list, in uBlock too, but later, due to lawsuits from website owners, against the owner of the POL list, due to the removal of anti-adblock mechanisms on web sites, the owner was forced to split the list into two separate and independent lists and to move the anti-adblock list into another / separate repository under the care of another person (who is Ukrainian by the way), that's why there are 2 separate POL filter lists.

But worth to mention, that there also exist more pairs of regional filters in uBO, the following threads contain discussions/explanations:
- https://github.com/uBlockOrigin/uBlock-issues/issues/301
- https://github.com/uBlockOrigin/uBlock-issues/issues/1336

> serhiyguryev : Unfortunately, this option won't work for Manifest V3, because, as far as I understand, lists need to be added manually when developing the extension itself. So we need to come up with something else.

Custom filters are possible in MV3 (for example AdGuard), but it's out of uBOL philosophy scope:
- https://github.com/uBlockOrigin/uBOL-home/discussions/153

**Comment by gorhill** [comment]
@serhiyguryev Which exact URL should be used for a UKR list?

**Comment by serhiyguryev** [comment]
> @serhiyguryev Which exact URL should be used for a UKR list?

If I understand correctly, you are referring to our collection of filters? If so, the URLs for the filters are as follows.

GitHub repository:
`https://github.com/ukrainianfilters/lists`

Ukrainian Ad Filter:
`https://raw.githubusercontent.com/ukrainianfilters/lists/main/ads/ads.txt`

Ukrainian Annoyance Filter:
`https://raw.githubusercontent.com/ukrainianfilters/lists/main/annoyances/annoyances.txt`

Ukrainian Privacy Filter:
`https://raw.githubusercontent.com/ukrainianfilters/lists/main/privacy/privacy.txt`

Ukrainian Security Filter:
`https://raw.githubusercontent.com/braveinnovators/ukrainian-security-filter/main/lists/adblock.txt`

**Comment by serhiyguryev** [comment]
> After the first install, uBO enables the appropriate regional list based on the browser language

@MasterKia On the one hand, as a developer, I understand this approach. However, I use exclusively English-language interfaces (operating system, browser, and other software), so this approach won't work in such cases.

**Comment by gorhill** [comment]
I will add only one list, so if you want to can create a `.txt` with `!#include ` statements if you want to combine ads/privacy/security while keeping them in separate files. I don't think annoyances should be combined in the list which will be enabled by default. A list combining ads/privacy/security would match uBO's own defaults.

I looked at `https://raw.githubusercontent.com/ukrainianfilters/lists/main/privacy/privacy.txt`, and I see there are a lot of cosmetic filters to remove comment sections -- why? Typically cosmetic filters have no privacy value as they act on the DOM rather than on the network requests, we can't assume no network request will occur when modifying the DOM. Also, removing comment sections is likely to be seen as breakage by end users, we do not do this in default lists, only in annoyance lists.

**Comment by stephenhawk8054** [comment]
```adb
||disqus.com^$third-party
```

Yeah, this will break all disqus comments in non-Ukrainian sites too.

We are the ones who will receive the reported issues from users. And when regional filter list authors have different policies, we will have to override the filters ourselves, causing more work for other volunteers.

**Comment by serhiyguryev** [comment]
> If we take the [AdGuard Ukrainian filter](https://filters.adtidy.org/extension/chromium/filters/23.txt) as an example, we will find more than 8 thousand rules in it. This is a very fresh filter, it was created by Ukrainians for Ukrainians in March 2023

Then why does this list contain 1700+ rules for .ru / .рф domains, 60+ for .by domains, 120+ for .kz domains?

> There is no difference between a charitable foundation or a support fund if the foundation's banner is placed on a third-party site.

For Russians, there may be no difference, but for our team there is. Therefore, when creating rules, we make [special (rules) exceptions](https://mastodon.online/@serhiyguryev/112954189339094220) for banner campaigns of Ukrainian charitable foundations. Moreover, in the case I mentioned, we are talking about **the targeted hiding of banners** of Ukrainian charitable foundations.

> Thus, we see how two different teams (AdGuard and RU AdList) with many years of experience

Then why don't yours and AdGuard's filters remove ads on popular Ukrainian websites (I've given examples)?

**Comment by serhiyguryev** [comment]
> I looked at `https://raw.githubusercontent.com/ukrainianfilters/lists/main/privacy/privacy.txt`, and I see there are a lot of cosmetic filters to remove comment sections -- why?

Through commenting systems built into Ukrainian popular websites, links to phishing sites that steal personal data and access to online bank account management systems are being massively distributed. If this is the only criticism of this filter, we can move these rules to a more specialized filter - Ukrainian Security Filter

**Comment by serhiyguryev** [comment]
> Lots of unnecessary wildcards in `https://raw.githubusercontent.com/braveinnovators/ukrainian-security-filter/main/lists/adblock.txt`

The motivation for creating such rules is disclosed in the filter description.

After Russia's full-scale invasion of Ukraine, the number of phishing campaigns has grown exponentially. Many of these campaigns are organized and managed from the territory of Russia. And this is a huge problem that can be compared to an epidemic but in the digital dimension, as Ukrainians lose their last money by falling for phishing websites.

Every day, dozens of new domain names are registered, which will be used in new phishing campaigns and online fraud. In such circumstances, blocking malicious web resources by domain name alone is not effective. 

That is why the Ukrainian Security Filter contains three groups of filtering rules: 

1) domain-based blocklist

2) page-specific filtering rules (blocking pages in social networks, links to groups or chatbots in messengers, etc.) 

3) pattern-based filtering rules (blocking is based on patterns).

What you have written about is pattern-based filtering rules that proactively block the phishing patterns we have detected (at the level of domain names, at the level of html page code, and at the level of network requests).

If you can come up with a more sophisticated approach that will help us fight dozens of new phishing campaigns every day and hundreds of new domain names used in them, I'd love to read about it.

**Comment by stephenhawk8054** [comment]
> No problem, we can remove this rule so that it does not conflict with other filters

I mean, it's just an example. We are very hesitant to consider regional lists that could potentially block connections used by other international sites. We already have EasyList and EasyPrivacy for these types of filters. 

How the lists are written is up to lists' authors, but we need to assure our own policies to balance usability and performance for users in general. Otherwise, it's hard for us to consider using that list, unless there are no other choices.

**Comment by serhiyguryev** [comment]
> If you use more than two wildcards, then the filter is evaluated internally as a regex (slower) instead of pattern (faster). Use wildcards sparingly.

Thanks for the advice, we will fix it

**Comment by stephenhawk8054** [comment]
If I'm not wrong, more than 1 wildcard is already making the filters to become regex.

**Comment by gorhill** [comment]
Right, these filters:

    ://*.*/user.php?mod=do&act=register&fromto=*$document,xmlhttprequest
    ://*.*/user.php?mod=do&act=login&fromto=*$document,xmlhttprequest
    ://*.*/user.php?mod=do&act=login$document,xmlhttprequest
    ://*.*/user.php?mod=do&act=register$document,xmlhttprequest

To:

    /user.php?mod=do&act=login$document,xmlhttprequest
    /user.php?mod=do&act=register$document,xmlhttprequest

And any filters in the form:

    ://*loto*.*/comission.php$document

Can be better rewritten as:

    *loto*.*/comission.php$document

And filters such as:

    ://*.*/*0registrations.html$document

Can be written as:

    *0registrations.html$document

Because the `://` part will always match, there is no point using it in a pattern when followed by a wildcard, all network-related URLs have a `://` part in them.

No filters should cause widespread breakage outside their region, so having a filter such as `||disqus.com^` in a list is a red flag, unless it is in an annoyance lists making its purpose clear.

But there are other concerns, it's not just the quantity in RUAdList, it's also the quality of the filters, and additionally there are a lot of filters in RUAdList which uses advanced filtering capabilities such as scriptlet injections, procedural cosmetic filtering, etc. Bottomline is that RUAdList is very mature filter list from maintainers with a lot of experience in crafting optimal filters -- this is what we expect to make it as a stock list in uBO.

**Comment by gorhill** [comment]
> If I'm not wrong, more than 1 wildcard is already making the filters to become regex.

A single wildcard which can't be ignored will cause a regex to be used internally.

**Comment by gorhill** [comment]
Added to dev version of stock lists for the time being.

**Comment by dportvine** [comment]
> Added to dev version of stock lists for the time being.

I thought you would add it to regional filters and not replace RU AdList. It will not be possible to completely replace it so quickly, and after removing the Ukraine region, in stable version of uBO/uBOL users will lose support from RU AdList. It is better not to hurry.

**Comment by gorhill** [comment]
> in stable version of uBO/uBOL users will lose support from RU AdList

I do not plan to change stable releases for a while, and anyway people can still enable RUAdList, it's not being removed from stock lists.

We have this statement in our README:

> New filter list requests are not permitted and will be declined and closed.

To be honest I hate to deal with such issue of people arguing endlessly that their list should be a stock lists, we have gone through this many times, hence why the request to not submit lists. In the end I made the change here against what I think is best given the tone of the thread, so I gave in because I don't have the time and motivation to discuss this forever. If in the end users have issues with the UKR list, best is to discuss it with the issue tracker of the list maintainers, not here.

**Comment by myroslavandriychuk** [comment]
@gorhill we have taken the previous comments into account and created a combined filter list (ads + privacy + security) - Ukrainian Filters (uBlock Origin Combined List) that we will continue to work with. 

If you have no objections to this list, please add it instead of ads.txt: `https://raw.githubusercontent.com/ukrainianfilters/lists/main/combined/uBO/uBO.txt`

Thank you in advance

**Comment by stephenhawk8054** [comment]
The overuse of wildcard in Ukrainian list is causing a lot of false positives:

- https://github.com/uBlockOrigin/uAssets/issues/28155
- https://github.com/uBlockOrigin/uAssets/issues/28156

Those filters are too generic. `mono.*.$document` is simply unacceptable.

I'm seriously doubting the quality of this list to be considered in uBO.

**Comment by serhiyguryev** [comment]
@stephenhawk8054 please, suggest a better option: https://github.com/uBlockOrigin/uBlock-issues/issues/2692#issuecomment-2539460715

The case of the rule you mentioned is not “too generic” because this rule blocks a dozen patterns used in phishing campaigns targeting Ukrainian citizens.

In this case, I don't care that the user didn't open something on the russian .su domain (although this is only true for uBOL). What matters to me is that our rules provided basic protection and saved money for my compatriots.

In general, our rules have blocked hundreds of phishing campaigns and we have not received an excessive number of complaints from Ukrainian users about false positives (the few cases that did occur were promptly corrected).

I am also surprised that messages from one user are used as a reason to remove our filters. Especially since the cases concern websites in the russian .su zone and the Azerbaijani .az zone (https://github.com/ukrainianfilters/lists/issues/258). Even the screenshot of the web browser that this user left on our Issues page also has a russian interface for some reason. This is probably a coincidence, since the phishing campaigns against which the rules are created are launched mainly from russia.

![Image](https://github.com/user-attachments/assets/67db4373-b762-4de4-a113-1bde3bc279ae)

You write that Adguard Ukrainian and RuAdlist are supported by more professional maintainers, so why don't these lists block ads on popular Ukrainian resources (as I wrote about above)?

The way I see the solution to the problem of russian users having problems with Ukrainian filters. We either ignore this or remove the block of pattern-based filtering rules from the SECURITY FILTERS section, depriving users from Ukraine of basic protection against phishing attacks.

**Comment by stephenhawk8054** [comment]
In uBO's lists, we don't try to block all kinds of matched patterns at all costs. Depending on the situations, we will try to narrow down the filters as much as possible. If there are no ways than including just a simple string that can potentially block many other URLs, we will drop that option. We don't try to block all of the internet, or punish every website that uses `mono` as their subdomains. Blocking all malicious links on the internet is impossible.

That's our assessments and criteria for each list to be considered in uBO. If you have your own criteria, we will respect that and choose other suitable lists.

> more professional maintainers

I didn't talk about `professional maintainers` or anything. I talked about the lists are maintained by more experienced maintainers, in terms of crafting the filters and trying to avoid false positives.

> In this case, I don't care that the user didn't open something on the russian .su domain (although this is only true for uBOL). What matters to me is that our rules provided basic protection and saved money for my compatriots.

Yeah, then I agree that this is not aligned with uBO's assessment, and it's better to not include the list in uBO. We always try to balance the filters, focusing too much on blocking and ignoring false positives completely is not suitable for uBO.

**Comment by serhiyguryev** [comment]
> We don't try to block all websites on the internet, or punish every website that uses mono as their subdomains

Just like non-Ukrainian citizens will not activate our filter.


Just to understand the situation. Do you personally dislike our list and, as a member of the development team, have already made the final decision to remove it, despite the fact that there are no complaints from Ukrainian users?

**Comment by serhiyguryev** [comment]
@stephenhawk8054 Above I have proposed a solution. We can remove the pattern-based filtering rules from the SECURITY FILTERS section right now. Then I will explain this decision to Ukrainian users that these rules go against the policy of your extension.

However, I ask you to explain then whether politically motivated blocking, which is in the RuAdlist created by "more experienced maintainers", does not contradict your rules:
||2ip.ua/images/support_ukraine.png
||2ip.ua/images/war.png

**Comment by mapx-** [comment]
@serhiyguryev 

what do you want to block using this filter:

```adb
mono.*.$document
```

**Comment by stephenhawk8054** [comment]
> Do you personally dislike our list

What is this question? I'm simply assessing based on the filters included in the lists. This is not the first time in this thread alone we were concerned with how the filters are written in this list:

- https://github.com/uBlockOrigin/uBlock-issues/issues/2692#issuecomment-2539397800

- https://github.com/uBlockOrigin/uBlock-issues/issues/2692#issuecomment-2539430246

- https://github.com/uBlockOrigin/uBlock-issues/issues/2692#issuecomment-2539515315

We can't monitor and feedback every line of filter if every time there's another issue of filter quality occurring.

I also said above:

> I mean, it's just an example. My point is, we are very hesitant to consider regional lists that could potentially block connections used by other international sites.

---

As said by gorhill, the tone of this thread really seems to go somewhere outside of what we are trying to focus: how the list written

> In the end I made the change here against what I think is best given the tone of the thread, so I gave in because I don't have the time and motivation to discuss this forever.

**Comment by serhiyguryev** [comment]
> We can't monitor and feedback every line of filter if every time there's another issue of filter quality occurring

@stephenhawk8054 Please do not avoid answering a specific question. I've given you a specific situation, will you deal with it, or is everything okay for you? Isn't the example I gave you a violation of your rules?

Your references to previous comments are no longer relevant.

@gorhill I ask you to intervene in the situation. I have proposed a solution. Are you, as the lead developer, satisfied with it?

**Comment by stephenhawk8054** [comment]
> Please do not avoid answering a specific question. I've given you a specific situation, will you deal with it, or is everything okay for you?

As I said, it's not ok for me. 

> I mean, it's just an example. My point is, we are very hesitant to consider regional lists that could potentially block connections used by other international sites.

> We can't monitor and feedback every line of filter if every time there's another issue of filter quality occurring.

The previous references are to point out that the overblocking is a recurring issue of this list.

**Comment by serhiyguryev** [comment]
> The previous references are to point out that the overblocking is a recurring issue of this list
Where are the complaints from Ukrainian users of the list?

> As I said, it's not ok for me.
What do you plan to do with this next?

**Comment by stephenhawk8054** [comment]
> What do you plan to do with this next?

As I said

> Usually if a list has quality issue, after our feedback we have to wait some time and check the list again if some users want to include it again in uBO. Here we already bypass the due diligence to include the list really fast due to the tone of the thread, and the list clearly still has the same kind of issue.

it's better to not include the list now and check the filters in the list after some time. 

That is the normal procedure for all kinds of new lists proposed. It's not about just this list.

**Comment by last-partizan** [comment]
Hi.

Ukrainian user here. I will repeat/explain what was said by @serhiyguryev about blocking mono*

It is useful, because we often donate money for emergency/medical supplies/whatever, and we use Monobank. Fishing attacks targeting their domain are blocked by this filter.

I had 0 false positives, with this rule. I've had another false positive with some other rule, and it was resolved very fast.

Giving control over Ukrainian filter list to someone other than Ukrainians seems inaproppriate.

**Comment by f1xmAn** [comment]
> Giving control over Ukrainian filter list to someone other than Ukrainians seems inaproppriate.

I agree with that. Also, the quality of the list must be assessed by the target audience. I don't see any issues with this rule raised by Ukrainians.

**Comment by huyszn** [comment]
This filter in the Ukrainian Filters list:
```adb
*petition*.*/$document,to=~competitions.obozrevatel.com|~e-dem.ua|~gov.ua
```
blocks a lot of international sites like:
```
https://www.openpetition.eu/
https://petition.parliament.uk/
https://petitions.parliament.nz/
https://www.ipetitions.com/
```

**Comment by stephenhawk8054** [comment]
> the quality of the list must be assessed by the target audience.

We don't just assess the filters based on the reports alone. If the filters contain too generic strings with low further specific patterns, those would end up blocking a lot of unexpected different kinds of URLs. `mono`, `telegram`, `petition`... are just too generic words to be used alone in pattern-based filters:

```
https://withmono.vercel.app/blog/understanding-webhooks-mono
http://mono.github.io/website/archived/web_services/
https://www.aplus.net/
https://aplus.physio/
https://www.bankid.com/
```

**Comment by last-partizan** [comment]
Yeah, those `?traffic_source` rules indeed look like holes and needs to be fixed.

But, I don't think wildcard rules are useless. Because scammers often target our fundraisers, and having some protection is better than none. False positives can be reported and resolved by Ukrainian filters team. Or manually, it's just one click on "Don't warn me about this site.

But, having "Adguard Ukrainian" fitter for Ukraine is simply unacceptable. They are Russian company, and they are working for Russians.

Let me add some context, about rules mentioned by @serhiyguryev :

![War.png](https://2ip.ua/images/war.png)

These aren't ads. We use similar banners to promote fundraisers, or put them somewhere on our projects to raise awarennes about the war.

![Support Ukraine](https://2ip.ua/images/support_ukraine.png)

Come Back Alive is one of the most efficient Ukrainian charity fundations. And Russians want to silence them.

These are the rules from "RU Adlist".

![Image](https://github.com/user-attachments/assets/49bd5a45-bd4b-4021-88e2-a76f5f638a91)

```
||2ip.ua/images/support_ukraine.png
||2ip.ua/images/war.png
```

It makes sense to hide these banners for Russians who are supporting the war. And it seems like Adguard is the Russians supporting the war to me.

And even if they aren't and those rules was somehow appeared by itself, without malicious intent, there's no reason to allow Russians to manage Ukrainian filters, or to show Ukrainian flag on "RU Adlist" filter.

**Comment by stephenhawk8054** [comment]
> I don't think wildcard rules are useless.

Yes, I didn't talk about abandoning them completely. I said that don't overuse it with very simple strings that can potentially match with a variety of unexpected URLs. Use regex or other separators (`^`, `|`...) with clear examples and documentations of which URLs you want to match to narrow down as much as possible.

Overblocking will just result in more false positives, and as a consequence, you have to create more exceptions filters for it and potentially poke more holes to the whole blocking systems.

> Or manually, it's just one click on "Don't warn me about this site.

The problem is due to [the limitations of MV3](https://github.com/w3c/webextensions/issues/744), in uBOL, `strict-blocking` is only available by the filters in the form of `||example.com^` or `||example.com^$document`. Other pattern-based `document` blocking filters won't yield the strict-blocking page for users to choose.

We can only choose the lists that are compatible to both uBO and uBOL.

---

About the latter part, PLEASE

> Do not bring politics or geopolitics here, that is not how decision are made here. More of this and I will lock the discussion.

RuAdlist has already answered it:

> As for blocking advertising of third-party funds, it is carried out in accordance with [EasyList Policy](https://easylist.to/pages/policy.html). There is no difference between a charitable foundation or a support fund if the foundation's banner is placed on a third-party site. All such banners have been blocked by the RU AdList filter for the past 9 years and this will not change.

If you don't agree, discuss somewhere else with RuAdlist and EasyList about their policies.

In our policies, we have no interests in any kinds of politics discussion. We only talk about how the filters are written technically. If you continue to bring this up, this will just cement more to us that you prioritize this aspect than the filters' performance and breakages, and it will just confirm that your policies and priorities are even more further away from our priorities.

**Comment by last-partizan** [comment]
I understand and respect your desire not to deal with any political issues.

I would be happy not to deal with them either, but we cannot avoid it completely. As a user, I do not want to discuss RuAdlist policies with them, as I do not believe they have my best interests in mind nor do they speak my language. I just want to have Ukrainian filters from Ukrainians, and I can report problems to them if any.

So, please keep Ukrainian filter in Ublock.

Quoting @serhiyguryev 

> @stephenhawk8054 Above I have proposed a solution. We can remove the pattern-based filtering rules from the SECURITY FILTERS section right now. Then I will explain this decision to Ukrainian users that these rules go against the policy of your extension

### Option 1

If this is okay, let's do it. But maybe it's better to split the filter, into something like "Base" and "Extended Security", just make sure they both available, even if only fast base filter is enabled by default.

Phishing attacks are real for us, and cost us money. And team behind Ukrainian filter works hard to counter the threats.

Speaking about this:

https://github.com/uBlockOrigin/uAssets/issues/28156

If my understanding is correct, uBlock Origin handled filter fine, but uBlock Origin Lite for Chrome is having problems due to limited capabilities.

### Option 2

Maybe, the best technical solution is to make two versions of the list: one for uBlock Origin, and one for uBlock Origin Lite?

Speaking as a user, I would prefer having those anti-phishing rules on by default, because it's easier for me just to install extension on my parents computer, and not to worry about phishing.

**Comment by stephenhawk8054** [comment]
> If my understanding is correct, uBlock Origin handled filter fine, but uBlock Origin Lite for Chrome is having problems due to limited capabilities.

It's not always that uBO handled the filters fine, these filters

```adb
telegram*.*/$document,to=~t.me|~telegram.me|~telegram.org
*petition*.*/$document,to=~competitions.obozrevatel.com|~e-dem.ua|~gov.ua
*aplus*.*/$document,to=~gov.ua
*bankid*.*/$document,to=~accordbank.com.ua|~altbank.ua|~asviobank.ua|~a-bank.com.ua|~bank.com.ua|~bank34.ua|~bankalliance.ua|~banklviv.com|~bankvostok.com.ua|~bisbank.com.ua|~cib.com.ua|~clearinghouse.com.ua|~creditdnepr.com|~credit-agricole.ua|~crystalbank.com.ua|~eximb.com|~globusbank.ua|~gov.ua|~grant.ua|~ideaonline.ua|~industrialbank.ua|~izibank.com.ua|~kredobank.com.ua|~monobank.com.ua|~monobank.ua|~motor-bank.ua|~mtb.ua|~oschadbank.ua|~otpbank.com.ua|~piraeusbank.ua|~poltavabank.com|~pravex.com.ua|~privat24.ua|~privatbank.ua|~pumb.ua|~radabank.com.ua|~raiffeisen.ua|~rwsbank.com.ua|~sensebank.com.ua|~tascombank.ua|~ukrgasbank.com|~ukrsibbank.com|~unexbank.ua|~vostok.bank
```

causes these kinds of sites being blocked in uBO as well:

```
https://github.com/telegramdesktop/tdesktop/blob/dev/.github/CONTRIBUTING.md
https://www.openpetition.eu/
https://petition.parliament.uk/
https://petitions.parliament.nz/
https://www.ipetitions.com/
https://www.aplus.net/
https://aplus.physio/
https://www.bankid.com/
```

I know that one of them is already addressed, but my point is not about that. My point has always been about:

> My point is, we are very hesitant to consider regional lists that could potentially block connections used by other international sites.

I'm very sorry to say this, but currently I have very low confidence of how effective the filters are being crafted in the list right now. Crafting the filters is not about just throwing the matching strings and put the wildcards everywhere else. I'm not sure even when this issue is resolved, a similar situation would occur again or not.

Currently all pattern-based filters that include very common strings need to be revised, regardless of uBO or uBOL.

---

I also see these generic cosmetic filters:

```adb
##html > body:has(div.popup > div.popup-inner > a[href*="/vote"])
##html > body:has(nav.header__nav > a.header__login[href*="/vote"])
##html > body:has(form[action="/vote"] > button[name="send_phone"])
##html > body:has(form[action="/vote/code"] > button[name="send_code"])
```

Not all CSS selectors have the same performance, [`:has()` is very expensive for large elements like `body`](https://github.com/refined-github/refined-github/issues/5797#issuecomment-2148865125). Using `##html > body:has()` as generic cosmetic filters won't have good performance since the surveyors for these filters must work on all sites.

**Comment by myroslavandriychuk** [comment]
@stephenhawk8054, as @serhiyguryev has already mentioned, we can remove pattern-based filtering rules from the SECURITY FILTERS section, creating a solution for Ukrainian users that will not depend on your extension.

But in that case, our team would like to know whether you will remove our filters based on your own beliefs, or if this issue will be resolved at this point?

**Comment by stephenhawk8054** [comment]
It will depend on other volunteers' opinions as well. This is the most I can feedback and give opinions about this list's filters.

**Comment by myroslavandriychuk** [comment]
> It will depend on other volunteers' opinions as well. This is the most I can feedback and give my opinions about this list's filters.

At the same time, the opinion of Ukrainian users, who have had only a few problems since testing this list in uBoL, which were quickly resolved, seems to be of no importance?

**Comment by serhiyguryev** [comment]
In order not to deprive Ukrainian users of the opportunity to have high-quality filters that actually remove ads from Ukrainian websites, the PATTERN-BASED FILTERING RULES section has been removed from the list. To protect against phishing, we will create a solution that does not depend on your extension.

Open source software should give you the right to choose. Currently, Ukrainian users have it. And this is important to our team.
